activitysim/abm/models/vehicle_type_choice.py

```diff
@@ -0,0 +1,236 @@
+# ActivitySim
+# See full license in LICENSE.txt.
+
+import logging
+
+import pandas as pd
+import numpy as np
+import itertools
+import os
+
+from activitysim.core.interaction_simulate import interaction_simulate
+from activitysim.core import simulate
+from activitysim.core import tracing
+from activitysim.core import config
+from activitysim.core import inject
+from activitysim.core import pipeline
+from activitysim.core import expressions
+from activitysim.core import logit
+from activitysim.core import assign
+from activitysim.core import los
+
+from activitysim.core.util import assign_in_place
+
+from .util.mode import mode_choice_simulate
+from .util import estimation
+
+
+logger = logging.getLogger(__name__)
+
+
+@inject.step()
+def vehicle_type_choice(
+        persons,
```

Comment:
The way activitysim works, these method inputs are never actually specified by the user. The `inject()` decorator instead tells the function to read those inputs from the activitysim "pipeline" object. So a given user never really needs to concern themselves with what these individual inputs actually are, or their types. Thus none of the other activitysim "steps" have type hints. But still, if you think it would be useful, I can add them.

Reply:
We don't use type hints anywhere else in ActivitySim, and I vote we don't suddenly start.  Personally I find they tend to be more confusing than helpful for many non-expert Python coders, as they look like code but they're just "hints" and not actually enforced by Python.  Personally I prefer to see types listed in docstrings, which look like comments and are more clearly not code used by Python, but for which any decent IDE can parse and support for linting.  It'd be great if we maintained proper docstrings in ActivitySim, not just for user-facing functions but in every non-trivial function everywhere, as it's super helpful for developers.